connectors/sources/generic_database.py

```diff
@@ -0,0 +1,444 @@
+import asyncio
+from datetime import date, datetime
+from decimal import Decimal
+from functools import partial
+
+from asyncpg.exceptions._base import InternalClientError
+from bson import Decimal128
+from sqlalchemy import text
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource
+from connectors.utils import iso_utc
+
+DEFAULT_FETCH_SIZE = 50
+DEFAULT_RETRY_COUNT = 3
+DEFAULT_WAIT_MULTIPLIER = 2
+DEFAULT_SSL_DISABLED = True
+DEFAULT_SSL_CA = None
+
+
+class GenericBaseDataSource(BaseDataSource):
+    """Class contains common functionalities for Generic Database connector"""
+
+    def __init__(self, connector):
+        """Setup connection to the database-server configured by user
+
+        Args:
+            connector (BYOConnector): Object of the BYOConnector class
+        """
+        super().__init__(connector=connector)
+
+        # Connector configurations
+        self.retry_count = int(
```

Comment:
with the latest changes, you can do:

```
self.retry_count = self.configuration['retry_count']
```

The base class is taking care of the default value and the type conversion now, as long as you set the right type in `get_default_configuration`

Reply:
Done